provision/aws/rosa_delete_cluster.sh

```diff
@@ -25,9 +25,14 @@ if [ -z "$REGION" ]; then echo "Variable REGION needs to be set."; exit 1; fi
 
 cd ${SCRIPT_DIR}/../opentofu/modules/rosa/hcp
 WORKSPACE=${CLUSTER_NAME}-${REGION}
-./../../../destroy.sh ${WORKSPACE}
+./../../../destroy.sh ${WORKSPACE} &
+DESTROY_PROCESS_ID=$!
 
 LOG_DIR="${SCRIPT_DIR}/logs/${CLUSTER_NAME}"
 mkdir -p ${LOG_DIR}
 cd ${LOG_DIR}
-rosa logs uninstall --debug -c ${CLUSTER_NAME} > "$(custom_date)_delete-cluster.log"
+echo "Saving Uninstall Logs."
+rosa logs uninstall -c ${CLUSTER_NAME} --watch --debug > "$(custom_date)_delete-cluster.log" &
+LOGS_SAVING_PROC_ID=$!
+wait $DESTROY_PROCESS_ID $LOGS_SAVING_PROC_ID
```

Comment:
```suggestion
timeout 30m bash -c 'wait $DESTROY_PROCESS_ID $LOGS_SAVING_PROC_ID'

if [[ $? -eq 124 ]]; then
    echo "Timeout occurred after 30 minutes."
else
    echo "Processes completed without a timeout."
fi
```

Reply:
Thanks @kami619 for the suggestion. I have applied it.  Theoretically I am not sure if the processes will stuck, but I agree for being on the safe side it worth to have a timeout there. Usually the process lasts ~18-20 minutes, so 30min timeout I think is fine.